### PR TITLE
Clean up usage of null safety features for gist logic

### PR DESCRIPTION
**Pull request by parlough**

### CODE REVIEW

File: lib/playground.dart [hunk 1]
```diff
@@ -687,9 +687,9 @@ class Playground extends EditorUi implements GistContainer, GistController {
 
     // When sharing, we have to pipe the returned (created) gist through the
     // routing library to update the url properly.
-    if (_overrideNextRouteGist != null &&
-        _overrideNextRouteGist!.id == gistId) {
-      editableGist.setBackingGist(_overrideNextRouteGist);
+    final overrideGist = _overrideNextRouteGist;
```
Comment (RedBrogdon) [inline lib/playground.dart:690]: I'm going to have to get used to seeing more local variables used strictly for null safety promotion. It took me a second to figure out what you were doing here.

**Review by RedBrogdon — APPROVED** [review]

File: lib/sharing/mutable_gist.dart [hunk 2]
```diff
@@ -12,7 +12,7 @@ import 'gists.dart';
 /// On overlay on a gist. Used to edit gists, this overlay knows about its dirty
```
[the remark this answers is not in the document]
Comment (parlough) [reply lib/sharing/mutable_gist.dart]: Good catch, fixed!

File: lib/playground.dart (hunk 1 above)
Comment (parlough) [reply lib/playground.dart:690]: Yeah, it's not everyone's cup of tea, but I find it generally improves readability and technically safety as well.

Kathy and Paul's article on [Fixing type promotion failures](https://dart.dev/tools/non-promotion-reasons) is a great reference to send to developers on how to best solve these types of issues.

**Comment by parlough** [comment]
@johnpryan I've merged the main branch in and fixed the resulting conflicts. Could you take a look when you get a chance? Thanks!

File: lib/sharing/mutable_gist.dart [hunk 3]
```diff
@@ -82,12 +82,16 @@ class MutableGist implements PropertyOwner {
...
-    }
-    if (summary != null) gist.summary = summary;
+    var gist = Gist(
+        description: description,
+        id: id,
+        public: public,
+        htmlUrl: htmlUrl,
+        summary: summary,
+        files: [
+          for (var file in getFiles())
```
Comment (johnpryan) [inline lib/sharing/mutable_gist.dart:92]: Much nicer!

**Review by johnpryan — APPROVED** [review]

**Pull request merged**